src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -813,17 +812,23 @@ class SnykToolWindowPanel(val project: Project) : JPanel(), Disposable {
     private fun displayNoVulnerabilitiesMessage() {
         descriptionPanel.removeAll()
 
-        val emptyStatePanel = JPanel()
-
-        emptyStatePanel.add(JLabel("Scan your project for security vulnerabilities and code issues. "))
-
-        val runScanLinkLabel = LinkLabel.create("Run scan") {
-            triggerScan()
-        }
+        val selectedTreeNode = vulnerabilitiesTree.selectionPath?.lastPathComponent
+        val messageHtmlText =
+            if (selectedTreeNode is RootContainerIssuesTreeNode) {
```

Comment:
Idea: Why not have an interface that defines a method to get the text. That way, each node could implement it and it would be nicer encapsulated and avoid then when/if/switch.